### PR TITLE
x1089 As TS (Katy) I would like to add which xenium machine is runnin…

### PR DESCRIPTION
**Pull request by sabrine33**

### CODE REVIEW

File: src/pages/XeniumAnalyser.tsx [hunk 1]
```diff
@@ -47,6 +49,7 @@ type LabwareSamples = {
 export type XeniumAnalyserFormValues = {
   lotNumberA: string;
   lotNumberB: string;
+  equipmentId: number | undefined;
```
Comment (seenanair) [inline src/pages/XeniumAnalyser.tsx:52]: Just to note, if you want to make it in a more ts way  you can always use equipmentId?: number which is the same as number|undefined

File: src/pages/XeniumAnalyser.tsx [hunk 2]
```diff
@@ -103,6 +108,7 @@ const XeniumAnalyser = () => {
       .max(new Date(), 'Please select a date and time on or before current time')
       .required('Time is a required field')
       .label('Time'),
+    equipmentId: Yup.number().required().label('Equipment').required('Equipment is a required field'),
```
Comment (seenanair) [inline src/pages/XeniumAnalyser.tsx:111]: Why `equipmentId`  is optional in form data structure and a required field in validation?

File: tests/unit/pages/lab/xeniumAnalyser.spec.tsx [hunk 3]
```diff
@@ -139,6 +144,21 @@ describe('Xenium analyser', () => {
           });
         });
       });
+      describe('Equipment', () => {
+        it('should display an error message on blur when no equipment is selected', () => {
+          waitFor(async () => {
+            await selectFocusBlur('equipmentId');
+            expect(screen.getByText('Equipment is a required field')).toBeVisible();
```
Comment (seenanair) [inline tests/unit/pages/lab/xeniumAnalyser.spec.tsx:151]: Just to understand, is this a better way to use it or await method where you do an action in act, outside of waitFor?

File: src/pages/XeniumAnalyser.tsx (hunk 2 above)
Comment (sabrine33) [reply src/pages/XeniumAnalyser.tsx:111]: so we can init the form with empty equipmentId option

Comment (seenanair) [reply src/pages/XeniumAnalyser.tsx:111]: Ok

File: tests/unit/pages/lab/xeniumAnalyser.spec.tsx (hunk 3 above)
Comment (sabrine33) [reply tests/unit/pages/lab/xeniumAnalyser.spec.tsx:151]: It depends on the use case. In this scenario, we are waiting for elements to become visible, which can be contingent on an asynchronous operation. We need to wait for the elements to appear before taking action and asserting the results. An alternative approach is to use findBy with a promise, like this:
 _screen.findByText('equipmentId').then((el) => {
            el.blur(); 
            expect(screen.getByText('Equipment is a required field')).toBeVisible();
          });_
But also, _el.blur();_  is not advisable in most cases.  To ensure our tests closely mimic user behavior, it's better to use userEvent to interact with the element, as it provides a more realistic simulation of user actions."

**Review by seenanair — APPROVED** [review]

**Pull request merged**